test/integration/udpingress_test.go

```diff
@@ -151,6 +164,217 @@ func TestUDPIngressEssentials(t *testing.T) {
 	}, ingressWait, waitTick)
 }
 
+func TestUDPIngressTCPIngressCollision(t *testing.T) {
+	t.Parallel()
+	t.Log("locking TCP and UDP ports")
+	udpMutex.Lock()
+	tcpMutex.Lock()
+	defer func() {
+		t.Log("unlocking TCP and UDP ports")
+		tcpMutex.Unlock()
+		udpMutex.Unlock()
+	}()
```

Comment:
Again, I'm feeling pretty strongly we should just make a pool of ports available.

Reply:
conversation moved to https://github.com/Kong/kubernetes-ingress-controller/pull/1702#discussion_r689675013